### PR TITLE
Added api proxy to core

### PR DESCRIPTION
**Pull request by jfkonecn**
resolves oncokb/oncokb-pipeline#336

### CODE REVIEW

File: src/main/java/org/mskcc/oncokb/curation/web/rest/ApiProxy.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import org.springframework.http.converter.StringHttpMessageConverter;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.server.ResponseStatusException;
+
+@RestController
+@RequestMapping({ "/api", "/legacy-api" })
```
Comment (zhx828) [inline src/main/java/org/mskcc/oncokb/curation/web/rest/ApiProxy.java:23]: proxy all `/api` as well?

File: src/main/java/org/mskcc/oncokb/curation/service/ApiProxyService.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    public HttpHeaders prepareHttpHeaders(String contentType) {
+        HttpHeaders httpHeaders = new HttpHeaders();
+        if (contentType != null) {
+            httpHeaders.setContentType(MediaType.valueOf(contentType));
+        }
+
+        return httpHeaders;
+    }
+
+    public ResponseEntity<String> exchangeRequest(String body, HttpMethod method, String contentType, String apiRequest)
```
Comment (zhx828) [inline src/main/java/org/mskcc/oncokb/curation/service/ApiProxyService.java:44]: This method doesn't seem to be used.

**Review by zhx828 — CHANGES_REQUESTED** [review]
Should we get rid of the following files from the shared/api/generated folder?

- .gitignore
- .npmignore
- .openapi-generator
- .openapi-generator-ignore

File: src/main/java/org/mskcc/oncokb/curation/web/rest/ApiProxy.java (hunk 1 above)
Comment (jfkonecn) [reply src/main/java/org/mskcc/oncokb/curation/web/rest/ApiProxy.java:23]: I thought we wanted all of core's API's. Do we just want the /legacy-api?

File: src/main/java/org/mskcc/oncokb/curation/service/ApiProxyService.java (hunk 2 above)
Comment (jfkonecn) [reply src/main/java/org/mskcc/oncokb/curation/service/ApiProxyService.java:44]: I may have done some copy and pasting....
I'll take care of it.

File: .prettierrc [hunk 3]
```diff
@@ -1,5 +1,8 @@
 # Prettier configuration
 
+plugins:
+  - prettier-plugin-java
```
Comment (jfkonecn) [inline .prettierrc:4]: @zhx828 I was having this [issue](https://github.com/jhipster/prettier-java/issues/508) where prettier couldn't figure out which parser to use for java files so I forced it to use the java parser.

**Review by zhx828 — APPROVED** [review]
Thanks! Don't forget to squash before merging.

**Comment by zhx828** [comment]
@jfkonecn the CI test failed.

**Pull request merged**